sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -485,3 +486,93 @@ export type EntityStatus =
   | "Renaming"
   | "Restoring"
   | "Unknown";
+
+/**
+ * An executor for a Promise (or function that returns a Promise) that obeys both a timeout and an
+ * optional AbortSignal.
+ * @param timeoutMs - The number of milliseconds to allow before throwing an AMQP error with condition ServiceUnavailableError
+ * @param timeoutMessage - The message to place in the .description field for the thrown exception for Timeout.
+ * @param abortSignal - The abortSignal associated with containing operation.
+ * @param abortErrorMsg - The abort error message associated with containing operation.
+ * @param value - The value to be resolved with after a timeout of t milliseconds.
+ * @returns {Promise<T>} - Resolved promise
+ *
+ * @internal
+ * @ignore
+ */
+export function waitForTimeoutOrAbortOrResolve<T>(args: {
+  actionFn: () => Promise<T>;
+  timeoutMs: number;
+  timeoutMessage: string;
+  abortSignal?: AbortSignalLike;
+  abortMessage: string;
```

Comment:
Now that abortMesage is mandatory, it should swap places with abortSignal as we want the optional arguments to appear in the end

Reply:
Done.